pulsebot/bugzilla.py

```diff
@@ -11,91 +11,59 @@
-            raise BugzillaError()
+        return r.json()
 
+    def _check_error(self, method, path, bug_data):
         if "error" in bug_data:
-            logger.error(
-                f"Error occurred retrieving bug fields {bug_url} {bug_data['error']}"
+            raise BugzillaError(
+                f"Data Error: {method} {self._server}/{path} {bug_data['error']}"
             )
-            raise BugzillaError()
 
+    def get_fields(self, bug, fields):
+        path = f"rest/bug/{bug}"
+        bug_data = self._call(
+            "GET", path, params={"include_fields": ",".join(fields)}
+        )
+        self._check_error("GET", path, bug_data)
         return bug_data.get("bugs", [{}])[0]
 
     def get_comments(self, bug):
-        bug_url = "%s/rest/bug/%d/comment?include_fields=text" % (self._server, bug)
-
-        logger.info(f"get_comments: {bug_url}")
-        try:
-            r = self._session.get(bug_url)
-            r.raise_for_status()
-            bug_data = r.json()
-        except Exception:
-            logger.exception(
-                f"Error occurred retrieving comments for {bug_url} {r.status_code}"
-            )
-            raise BugzillaError()
-
-        if "error" in bug_data:
-            logger.error(
-                f"Error occurred retrieving comments for {bug_url} {bug_data['error']}"
-            )
-            raise BugzillaError()
-
+        path = f"rest/bug/{bug}/comment"
+        bug_data = self._call(
+            "GET", path, params={"include_fields": "text"}
+        )
+        self._check_error("GET", path, bug_data)
         comments = bug_data["bugs"].get("%d" % bug, {}).get("comments", [])
         return [c.get("text", "") for c in comments]
 
     def post_comment(self, bug, comment):
-        try:
-            post_url = "%s/rest/bug/%d/comment" % (self._server, bug)
-            logger.info(f"post_comment: {post_url}")
-            r = self._session.post(
-                post_url,
-                data={
-                    "comment": comment,
-                },
-            )
-            r.raise_for_status()
-        except Exception:
-            logger.exception(
-                f"Error occurred posting comment to {post_url} {r.status_code}"
-            )
-            raise BugzillaError()
+        self._call("POST", f"rest/bug/{bug}/comment", data={"comment": comment})
 
     def update_bug(self, bug, **kwargs):
-        try:
-            post_url = "%s/rest/bug/%d" % (self._server, bug)
-            logger.info(f"update_bug: {post_url}")
-            r = self._session.put(
-                post_url,
-                data=json.dumps(kwargs),
-                headers={"Content-Type": "application/json"},
-            )
-            r.raise_for_status()
-        except Exception:
-            logging.exception(f"Error occurred updating bug {post_url} {r.status_code}")
-            raise BugzillaError()
+        self._call("PUT", f"rest/bug/{bug}", data=json.dumps(kwargs))
```

Comment:
Just noticed this -- wanted to make sure that you intended to pass this as a string, and not as a dictionary? (`data` should be able to handle a dictionary as long as it is serializable).